Kajabi/sage-lib
# Sage 749/mw list item prop type errors

## Description
ListItem component has a few errors in PropTypes declaration that are leading to console errors. This PR addresses the following:
* `id` should use `PropTypes.oneOfType`
* `moreActions` should wrap existing hash with `PropTypes.shape()`

## Testing in `sage-lib`
1. Navigate to http://localhost:4100/?path=/docs/sage-list--default
2. Inspect the page and view the console
3. Ensure the following errors are no longer showing

![Screen Shot 2022-07-29 at 1 41 41 PM](https://user-images.githubusercontent.com/24237393/181824152-3ae7c5fa-f77a-4dcd-804d-314ed1dbc36a.png)

## Testing in `kajabi-products`

1. (**LOW**) Console error

## Related
https://kajabi.atlassian.net/browse/SAGE-749
